sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateTableAsSelect.scala

```diff
@@ -50,17 +50,24 @@ case class CreateTableAsSelect(
       import org.apache.hadoop.io.Text
       import org.apache.hadoop.mapred.TextInputFormat
 
-      val withSchema =
+      var withSchema =
```

Comment:
We we really need a var here?  Can't we just have `withFormat` followed by `withSchema`?